Assignment7/Assignment7_QNo3/SolAssignmentNo7_QNo3.rb

```diff
@@ -0,0 +1,57 @@
+# => 3. Write a program to describe two difference between proc and lambda.
+class ClassDemo
+  # => method to demonstrate the first difference between lambda and proc
+  def demo_first_diff
+    # => input two names
+    puts "Enter the first name"
+    fname = gets.chomp
+    puts "Enter the second name"
+    lname = gets.chomp
+
+    # => way the proc is created
+    full_name = Proc.new { |fname, lname| puts "Proc called :#{fname} #{lname}" }
+    # => call the proc
+    full_name.call(fname, lname)
+
+    # => way the lambda is created
+    full_name = -> (fname, lname) { puts "Lambda called : #{fname} #{lname}" }
+    # => call the lambda
+    full_name.call(fname, lname)
+  end
+
+  # => method to demonstrate the second difference between lambda and proc
+  def demo_second_diff
+    # => input two names
+    puts "Enter the first name"
+    fname = gets.chomp
+    puts "Enter the second name"
+    lname = gets.chomp
+
+    # => way the proc return from current context
+    full_name = Proc.new { |fname, lname| puts "Not bothered about arguments" }
```

Comment:
@HrisikeshSinghaChowdhury you did not use the keyword return here. There is a significant different between Proc and lambda how they behave if there is an return statement inside that block

Reply:
done